framework/src/service/LockSet.cpp

```diff
@@ -0,0 +1,49 @@
+/*=============================================================================
+  Library: CppMicroServices
+  Copyright (c) The CppMicroServices developers. See the COPYRIGHT
+  file at the top-level directory of this distribution and at
+  https://github.com/CppMicroServices/CppMicroServices/COPYRIGHT .
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+=============================================================================*/
+
+#include "LockSet.h"
+
+#ifdef _MSC_VER
+#    pragma warning(push)
+#    pragma warning(disable : 4355)
+#endif
+
+namespace cppmicroservices
+{
+
+#ifdef US_ENABLE_THREADING_SUPPORT
+    LockSet::LockSet(std::shared_ptr<ServiceRegistrationBasePrivate> reg,
+                     std::shared_ptr<ServiceRegistrationCoreInfo> coreInfo)
+    {
+        if (reg != nullptr)
+        {
+            regL = reg->Lock();
+        }
+        coreInfoL = coreInfo->Lock();
```

Comment:
`reg` is checked for `nullptr` but `coreInfoL` is not? should it be?

Reply:
`reg` is checked because reg is not owned by `ServiceReferenceBasePrivate` and so it can die with `ServiceReferenceBasePrivate` still with a `weak_ptr` to it. So it has to be checked to confirm that it still exists. In contrast, `coreInfo` is owned by both the `ServiceReferenceBasePrivate` and the `ServiceRegistrationBasePrivate` so it is guaranteed to be alive. That is why this check was made, but it would also not cause problems to check both for consistency. The only caveat to generalizing this function, i.e. either lock can be passed in as the first or second argument is then you will get into deadlocks where one call is made with `LockSet(reg, coreInfo)` and the other will be made with `LockSet(coreInfo, reg)`.